crates/sui-indexer/src/indexer.rs

```diff
@@ -108,6 +108,10 @@ impl Indexer {
             )
             .await?;
 
+        // Index protocol configs for protocol versions not yet in the db
+        let store_clone = store.clone();
+        spawn_monitored_task!(store_clone.persist_protocol_configs_and_feature_flags());
```

Comment:
Does this mean that protocol configs are filled by a parallel process (i.e. not adhering/contributing to the same watermark as the "main" indexing task? That may pose an issue for consistency (where someone issues a query for an epoch and its protocol config and gets one but not the other).

Reply:
I understand your concern. This issue primarily arises during the startup of an indexer instance. Specifically, the `persist_protocol_configs_and_feature_flags` call at epoch change time is not a parallel process. However, if a user makes a query right after the indexer restarts after being down for more than an epoch, they could encounter this problem.

I agree that the process should be blocking and completed before we start indexing everything else. I have attempted to implement this, but ran into an issue. In my setup, we read the chain identifier from the database in `persist_protocol_configs_and_feature_flags`. This identifier is the digest of the first checkpoint, which requires the indexing of the first checkpoint to be completed in the database. That's why we wait in that while loop for the other process to finish indexing checkpoint 0.

Ideally, the process should follow this sequential order:
- Index the first checkpoint, if it has not been indexed yet.
- Call `persist_protocol_configs_and_feature_flags`.
- Index the rest of the checkpoints.

One potential solution to alleviate this issue is to perform the process in parallel only if the watermark is 0. This way, we only encounter the consistency issue at the beginning with an empty database, which no user will ever encounter. What do you think?

Reply:
Is it possible to pass the chain identifier to the catch-up process in memory rather than having the job fetch it from the DB? That's one way I can think of to keep the reads consistent always while maintaining this read order.

Reply:
An alternative is to add a configuration parameter to the indexer that sets which chain it is supposed to be indexing for, which is then used to initialise the protocol config table. That is simpler to implement but then we'll need to update all the deployment configs, and it adds a layer of complexity for starting up the indexer as well, I suppose.

Reply:
The issue is not so much about it being out-of-date, it's about it being out-of-sync: They could query for an epoch and its protocol config and the protocol config not exist in the DB yet because it's being filled in parallel.

Reply:
I've changed the code to pass in the chain identifier, and to index protocol configs at the same time we index checkpoint 0, so no busy waiting or separate process anymore.